cmake/FindLibCURL.cmake

```diff
@@ -0,0 +1,84 @@
+if (WIN32)
+    option(VCPKG_DEPENDENCY_EXTERNAL_LIBCURL "Use an external version of the libcurl library" OFF)
+else()
+    option(VCPKG_DEPENDENCY_EXTERNAL_LIBCURL "Use an external version of the libcurl library" ON)
+endif()
+
+if(POLICY CMP0135)
+    cmake_policy(SET CMP0135 NEW)
+endif()
+
+if (VCPKG_DEPENDENCY_EXTERNAL_LIBCURL)
+    find_package(CURL REQUIRED)
+    return()
+endif()
+
+# This option exists to allow the URI to be replaced with a Microsoft-internal URI in official
+# builds which have restricted internet access; see azure-pipelines/signing.yml
+# Note that the SHA512 is the same, so vcpkg-tool contributors need not be concerned that we built
+# with different content.
+if(NOT VCPKG_LIBCURL_URL)
+    set(VCPKG_LIBCURL_URL "https://github.com/curl/curl/archive/refs/tags/curl-8_8_0.tar.gz")
+endif()
+
+include(FetchContent)
+FetchContent_Declare(
```

Comment:
Since curl is a nontrivial build should we consider trying to host ourselves rather than using FetchContent for this case? Our existing other dependencies don't really have meaningful *builds* which is why we never opened that can of worms.